hooks/src/index.js

```diff
@@ -26,15 +26,45 @@ let oldBeforeUnmount = options.unmount;
 const RAF_TIMEOUT = 100;
 let prevRaf;
 
+let mask = 0;
+let localIdCounter = 0;
+
 options._diff = vnode => {
+	if (typeof vnode.type === 'function' && !vnode._mask) {
+		vnode._mask =
+			vnode._parent && vnode._parent._mask
+				? vnode._parent && vnode._parent._mask + 1
+				: mask++;
+	}
 	currentComponent = null;
 	if (oldBeforeDiff) oldBeforeDiff(vnode);
 };
 
+options.diffed = vnode => {
```

Comment:
Also called in RTS https://github.com/preactjs/preact-render-to-string/blob/master/src/index.js#L222